tests/run/capturing.scala

```diff
@@ -0,0 +1,9 @@
+class MT(sf: MT => String) {
+  // `s` is retained as a field, but `sf` should not be.
+  val s = sf(this)
+}
+object Test extends App {
```

Comment:
I am still not sure why we use LegacyApp instead of App now. They seem to be equivalent.